runtime/compiler/optimizer/InterpreterEmulator.hpp

```diff
@@ -0,0 +1,272 @@
+         methodHandleIndex(methodHandleIndex),
+         mutableCallsiteIndex(mutableCallsiteIndex){}
+      virtual MutableCallsiteTargetOperand* asMutableCallsiteTargetOperand(){ return this;}
+      virtual TR::KnownObjectTable::Index getKnownObjectIndex(){ return methodHandleIndex; }
+      TR::KnownObjectTable::Index methodHandleIndex;
+      TR::KnownObjectTable::Index mutableCallsiteIndex;
+   };
+
+class InterpreterEmulator : public TR_ByteCodeIteratorWithState<TR_J9ByteCode, J9BCunknown, TR_J9ByteCodeIterator, Operand *>
+   {
+   typedef TR_ByteCodeIteratorWithState<TR_J9ByteCode, J9BCunknown, TR_J9ByteCodeIterator, Operand *> Base;
+
+   public:
+      InterpreterEmulator(
+            TR_CallTarget *calltarget,
+            TR::ResolvedMethodSymbol * methodSymbol,
+            TR_J9VMBase * fe,
+            TR::Compilation * comp,
+            TR_InlinerTracer *tracer,
+            TR_EstimateCodeSize *ecs)
+         : Base(methodSymbol, comp),
+           _calltarget(calltarget),
+           _tracer(tracer),
+           _ecs(ecs),
+           _iteratorWithState(false)
+         {
+         TR_J9ByteCodeIterator::initialize(static_cast<TR_ResolvedJ9Method *>(methodSymbol->getResolvedMethod()), fe);
+         _flags = NULL;
+         _stacks = NULL;
+         }
+      TR_InlinerTracer *tracer() { return _tracer; }
+      /* \brief Initialize data needed for looking for callsites
+       *
+       * \param blocks
+       *       blocks generated from bytecodes
+       *
+       * \param flags
+       *       flags with bits to indicate property of each bytecode. The flags are set by \ref TR_J9EstimateCodeSize::processBytecodeAndGenerateCFG.
+       *
+       * \param callSites
+       *       the call sites array to be filled in with callsites found
+       *
+       * \param cfg
+       *       CFG generated \ref TR_J9EstimateCodeSize::processBytecodeAndGenerateCFG from bytecodes
+       *
+       * \param recursionDepth
+       *       the depth of inlining layers
+       *
+       * \parm callstack
+       *       the call stack from the current inlined call target to the method being compiled
+       */
+      void prepareToFindAndCreateCallsites(TR::Block **blocks, flags8_t * flags, TR_CallSite ** callSites, TR::CFG* cfg, TR_ByteCodeInfo *newBCInfo, int32_t recursionDepth, TR_CallStack *callstack);
+
+      /*
+       * \brief look for calls in bytecodes and create callsites
+       *
+       * \param wasPeekingSuccessfull
+       *       indicate whether trees has been generated by peeking ilgen
+       *
+       * \param withState
+       *       whether the bytecode iteration should be with or without state
+       */
+      void findAndCreateCallsitesFromBytecodes(bool wasPeekingSuccessfull, bool withState);
+      void setBlocks(TR::Block **blocks) { _blocks = blocks; }
+      TR_StackMemory trStackMemory()            { return _trMemory; }
+      bool _nonColdCallExists;
+      bool _inlineableCallExists;
+
+      enum Flag
```

Comment:
Is there a better name for this flag? a description of what each flag means might be helpful since `isUnsanitizaeable` isn't immediately obvious...

Reply:
changed the flag name to `BytecodePropertyFlag` and added a description for each flag. `isUnsanitizaeable ` is a concept in partial inlining but I am not very sure what doesn't it mean either.